pkg/gateway/middleware/subdomain.go

```diff
@@ -98,14 +95,30 @@ type SubdomainFields struct {
 	Name      string
 	Version   string
 	StubId    string
-	StubGroup string
+	Subdomain string
 }
 
 func parseSubdomain(host, baseDomain string) string {
-	domain := "." + baseDomain
-	if strings.HasSuffix(host, domain) {
-		return strings.TrimSuffix(host, domain)
+	// Remove port if present
+	h, _, err := net.SplitHostPort(host)
+	if err != nil {
+		h = host // If error, assume no port
 	}
+
+	// Normalize to lower case
+	normalizedHost := strings.ToLower(h)
+	normalizedBaseDomain := strings.ToLower(baseDomain)
+
+	// If host equals baseDomain, return empty string
+	if normalizedHost == normalizedBaseDomain {
```

Comment:
i assume this is for path routing instead of subdomain routing.

Reply:
Yes. If there's no difference between these two, then there isn't a subdomain to parse.